src/arduino-timer.h

```diff
@@ -131,10 +131,14 @@ class Timer {
                 const unsigned long duration = t - task->start;
 
                 if (duration >= task->expires) {
-                    task->repeat = task->handler(task->opaque) && task->repeat;
-
-                    if (task->repeat) task->start = t;
-                    else remove(task);
+                    Task activeId = task_id(task);
```

Comment:
By caching the task's `task_id(task)` and `task->repeat` states, it becomes safe to ignore whether the handler deletes the task while it is running.